google-cloud-logging/src/main/java/com/google/cloud/logging/MonitoredResourceUtil.java

```diff
@@ -161,6 +179,17 @@ private static String getValue(Label label) {
       case NamespaceId:
         value = MetadataConfig.getNamespaceId();
         break;
+      case NamespaceName:
+        try {
+          value =
+              new String(
+                  Files.readAllBytes(
+                      Paths.get("/var/run/secrets/kubernetes.io/serviceaccount/namespace")));
```

Comment:
1) Use encoding when create String from bytes.
2) User must be able to specify alternative path either with an Environment variable (K8S_NAMESPACE) or with a system property (like -Dk8n.namespace) or with both.

Reply:
Done